addon/components/connect.js

```diff
@@ -1,98 +1,113 @@
 import Ember from 'ember';
 
 const {
+  assert,
   computed,
   defineProperty,
+  getProperties,
   inject: { service },
-  isEmpty,
   run
 } = Ember;
 
-export default (stateToComputed, dispatchToActions=() => ({})) => {
+/**
+  Returns a list of keys that have different values between the two objects.
+
+  @method changedKeys
+  @return {Array} keys that have changed
+  @private
+*/
+function changedKeys(state, newState) {
+  return Object.keys(state).filter(key => {
+    return state[key] !== newState[key];
```

Comment:
I assume this is a 1 for 1 replacement of the work we did previously in `handleChange` ?

```js
        const notifyProperties = Object.keys(props).filter(name => {
          return this.get(name) !== props[name];
        });
```